homeassistant/components/livebox/config_flow.py

```diff
@@ -0,0 +1,151 @@
+from aiosysbus.exceptions import AuthorizationError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.core import callback
+
+from .const import (
+    CONF_LAN_TRACKING,
+    DEFAULT_HOST,
+    DEFAULT_LAN_TRACKING,
+    DEFAULT_PORT,
+    DEFAULT_USERNAME,
+    DOMAIN,
+    TEMPLATE_SENSOR,
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): str,
+        vol.Required(CONF_USERNAME, default=DEFAULT_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LiveboxFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Livebox config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the Livebox flow."""
+
+        self._session = None
+        self.host = None
+        self.port = None
+        self.username = None
+        self.password = None
+        self.box_id = None
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+
+        errors = {}
+        if user_input is not None:
+            self.host = user_input["host"]
+            self.port = user_input["port"]
+            self.username = user_input["username"]
+            self.password = user_input["password"]
+            try:
+                self._session = Sysbus(
+                    username=self.username,
+                    password=self.password,
+                    host=self.host,
+                    port=self.port,
+                )
+
+                perms = await self._session.async_get_permissions()
+                if perms is not None:
+                    return await self.async_step_register()
+
+            except AuthorizationError:
+                errors["base"] = "login_inccorect"
+
+        # If there was no user input, do not show the errors.
+        if user_input is None:
+            errors = {}
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_register(self, user_input=None):
+        """Step for register component."""
+
+        errors = {}
+        infos = await self._session.system.get_deviceinfo()
+        self.box_id = infos.get("status").get("SerialNumber")
+        entry_id = self.hass.config_entries.async_entries(DOMAIN)
+
+        if self.box_id is not None:
+            if entry_id and entry_id.get("data", {}).get("id", 0) == self.box_id:
+                self.hass.config_entries.async_remove(entry_id)
```

Comment:
Since we have a unique identifier here, we can set the unique ID on the configuration flow. This will automatically abort the configuration flow when a user tried to add the same box twice.